src/agent_fork/text.py

```
@@ -19,9 +19,36 @@
     "\r": r"\r",
 }
 
+# Bidirectional formatting characters reorder how text is displayed without
+# being control bytes, so the C0/C1 ranges below do not cover them. A branch
+# name carrying U+202E can make the rendered line read differently from the
+# name that is actually stored. `agents` imports this set for its own
+# terminal-safety predicate rather than keeping a second copy.
+BIDI_CONTROLS = frozenset(
+    {
+        "؜",
+        "‎",
```

Ruff (0.16.1)

[error] Contains control characters that can permit obfuscated code

(PLE2502)